solr/core/src/java/org/apache/solr/cluster/placement/plugins/AffinityPlacementFactory.java

```diff
@@ -171,14 +174,17 @@ public AffinityPlacementConfig getConfig() {
 
     private final long prioritizedFreeDiskGB;
 
+    private final Map<String, String> withCollections;
```

Comment:
Q: a given collection can only be `withCollection` for a single secondary collection? Doesn't seem necessary...

Suggestion: maintain the inverse mapping as well (a multimap, but possibly this one should be a multimap as well) to save looping through map keys checking values...

Reply:
There's no reason not to extend this to N:1 (N primary collections with 1 secondary), I just followed the model that was present in 8x.

However, the way this co-location is implemented now may degrade the placements if we allow N:1 or N:N models, because of the pre-filtering approach we could quickly run out of candidate nodes for placements. So I prefer to focus on implementing 1:1 now and making sure it works well.